src/components/molecules/ConfigurationCard/ConfigurationCard.tsx

```diff
@@ -77,10 +91,33 @@ const ConfigurationCard: React.FC<ConfigurationCardProps> = props => {
 
                 return (
                   <StyledFooterButtonsContainer>
-                    <Button onClick={onCancel} $customType="secondary" hidden={!onCancel || disabled}>
+                    <Button
+                      onClick={() => {
+                        onCancel?.();
+                        setError(null);
+                      }}
+                      $customType="secondary"
+                      hidden={!onCancel || disabled}
+                    >
                       Cancel
                     </Button>
-                    <Button onClick={onConfirm} $customType={isWarning ? 'warning' : 'primary'} disabled={disabled} hidden={!onConfirm}>
+                    <Button
+                      onClick={() => {
+                        validateFields?.().then(() => {
+                          const confirmResult = onConfirm?.();
+
+                          if (typeof confirmResult === 'object' && typeof confirmResult.then === 'function') {
+                            confirmResult.then((res: any) => {
+                              setError(res);
+                            });
+                          }
+                        });
```

Comment:
Not 100% sure, but it may be better to make `onConfirm` void, and allow setting `error` as props.

Beside, it rather doesn't make sense to handle `onConfirm`'s result as an error, rather `catch`

Reply:
I also was thinking about passing `error` as props, but then I will have to add `useState`s to every component using `ConfigurationCard`. So I think this approach is much better. Also helps to refactor `ConfigurationCard` later.

Sure, I can move handing to `catch`